youtube_dl/extractor/rtp.py

```diff
@@ -3,56 +3,100 @@
+    decoded_url = re.match(r"[^[]*\[([^]]*)\]", code).groups()[0]
+    return compat_b64decode(
+        compat_urllib_parse_unquote(
+            decoded_url.replace('"', '').replace('\'', '').replace(',', ''))).decode('utf-8')
 
 
 class RTPIE(InfoExtractor):
-    _VALID_URL = r'https?://(?:www\.)?rtp\.pt/play/p(?P<program_id>[0-9]+)/(?P<id>[^/?#]+)/?'
+    _VALID_URL = r'https?://(?:(?:(?:www\.)?rtp\.pt/play/(?P<subarea>.*/)?p(?P<program_id>[0-9]+)/(?P<episode_id>e[0-9]+/)?)|(?:arquivos\.rtp\.pt/conteudos/))(?P<id>[^/?#]+)/?'
     _TESTS = [{
-        'url': 'http://www.rtp.pt/play/p405/e174042/paixoes-cruzadas',
-        'md5': 'e736ce0c665e459ddb818546220b4ef8',
+        'url': 'https://www.rtp.pt/play/p9165/e562949/por-do-sol',
         'info_dict': {
-            'id': 'e174042',
-            'ext': 'mp3',
-            'title': 'Paixões Cruzadas',
-            'description': 'As paixões musicais de António Cartaxo e António Macedo',
+            'id': 'por-do-sol',
+            'ext': 'mp4',
+            'title': 'Pôr do Sol Episódio 1 - de 16 Ago 2021',
+            'description': 'Madalena Bourbon de Linhaça vive atormentada pelo segredo que esconde desde 1990. Matilde Bourbon de Linhaça sonha fugir com o seu amor proibido. O en',
             'thumbnail': r're:^https?://.*\.jpg',
         },
     }, {
-        'url': 'http://www.rtp.pt/play/p831/a-quimica-das-coisas',
+        'url': 'https://www.rtp.pt/play/p510/aleixo-fm',
         'only_matching': True,
     }]
 
     def _real_extract(self, url):
         video_id = self._match_id(url)
 
         webpage = self._download_webpage(url, video_id)
-        title = self._html_search_meta(
-            'twitter:title', webpage, display_name='title', fatal=True)
-
-        config = self._parse_json(self._search_regex(
-            r'(?s)RTPPlayer\(({.+?})\);', webpage,
-            'player config'), video_id, js_to_json)
-        file_url = config['file']
-        ext = determine_ext(file_url)
-        if ext == 'm3u8':
-            file_key = config.get('fileKey')
-            formats = self._extract_m3u8_formats(
-                file_url, video_id, 'mp4', 'm3u8_native',
-                m3u8_id='hls', fatal=file_key)
-            if file_key:
-                formats.append({
-                    'url': 'https://cdn-ondemand.rtp.pt' + file_key,
-                    'preference': 1,
-                })
+
+        # Remove JS multi-line comments from webpage source
+        webpage = re.sub(r'(\/\*.*\*\/)', '', webpage, flags=re.DOTALL)
+
+        title = self._html_search_regex(r'<title>(.+?)</title>', webpage, 'title')
+
+        # Replace irrelevant text in title
+        title = title.replace(' - RTP Play - RTP', '')
+
+        # Check if it's a video split in parts, if so add part number to title
+        part = self._html_search_regex(r'section\-parts.*<span.*>(.+?)</span>.*</ul>', webpage, 'part', default=None)
+        if part:
+            title = f'{title} {part}'
```

Comment:
This syntax seems to be specific to Python 3.6. Shouldn't the extractor code be compatible with 2.6 and 2.7 as well?

Reply:
Definitely. This wouldn't pass the CI tests. Have one of these instead.

Uncontroversial string operations:
```
            title = ' '.join((title, part, ))
```
```
            title += ' ' + part
```
Old-style format, still supported in Python3:
```
            title = '%s %s' % (title, part, )
```
```
            title = '%(title)s %(part)s' % {'title': title, 'part': part, }
```
```
            # if title and part are local vars ...
            title = '%(title)s %(part)s' % locals()
```
Newer-style, but still OK in 2.
```
            title = '{0} {1}'.format(title, part)
```
```
            title = '{title} {part}'.format(title=title, part=part)
```
You might get the impression that Guido couldn't make his mind up about how string formatting should work.